cwms-data-api/src/main/java/cwms/cda/api/RatingController.java

```diff
@@ -339,6 +338,10 @@ public void getAll(@NotNull Context ctx) {
                         + "otherwise specified), as well as the time zone of any times in the"
                         + " response. If this field is not specified, the default time zone "
                         + "of UTC shall be used."),
+                @OpenApiParam(name = EFFECTIVE_DATE, description = "Specifies the "
```

Comment:
This is a very confusing parameter. 
- three optional date parameters are confusing
- what happens when all three are input?
- the ask was for the current effective date, that means the client needs to always set this parameter to now. Why not just have another endpoint that does it automatically? /{rating-id}/latest-effective
- the "closest" effective date shouldn't return a newer effective date than what is supplied as the newer effective date is not yet in effect, though this is made more complicated by temporal interpolations and transition dates. 
- how do active ratings play an effect? I presume they would be skipped, but that's missing from the docs (even from the start/end parameters)

Additionally, you are passing this confusion into the DAO by expanding the existing DAO method instead of creating a new one, so now you have three dates in the DAO function. This makes test cases far more complicated.

Reply:
Moved functionality to its own endpoint to reduce confusion. New endpoint returns closest effective date rating where the date is not beyond the current date.